### DISCUSSION TITLE
DDOS Protection

### DISCUSSION THREAD

**Discussion by aftechro** [AzuraCast/AzuraCast#6983] [General]
Hey guys,

Does azuracast have an ddos protection app integrated?
What do you guys doing regarding the DDOS attacks? 

Thank you!

**Comment by BusterNeece** [comment]
DDoS protection typically takes place at a layer higher than the one our application occupies; for example, either at the host level for your provider, or at the reverse proxy level if you're using a service like Cloudflare (which you absolutely can use with AzuraCast, just with [some caveats](https://www.azuracast.com/docs/administration/cloudflare/).

The application itself doesn't take too many steps to mitigate DDoS traffic, other than the following:
 - Aggressively caching unauthenticated requests, particularly JSON and static assets, at the nginx level so multiple requests don't impact performacne
 - Rate-limiting on public-facing pages that are likely to be flooded, i.e. song requests and the login prompt

If you're concerned about possible DDoS attacks, I'd definitely recommend pursuing DDoS protection either through your VPS provider (many large ones like Vultr offer it as a service) or through Cloudflare.

**Comment by wppronto** [comment]
As the owner of a web hosting company, I echo what Buster says above. That normally happens at a level above AzuraCast. For example, we specialize in WordPress. Some of our clients want to WordPress plugins like Wordfence to fight things like DDoS attacks, but that isn't the way. That causes resource issues if multiple sites are running things like that. Instead, we show clients that we handle that at the server level, much more efficiently than the site level. We are also a Cloudflare Optimized Partner, so we recommend utilizing them where possible. Their free plan is more than enough for most people's needs.

**Comment by aftechro** [comment]
so i just need to check this with vps provider then, thought i can set something on the vps that runs the azuracast. 
thank you!
